### PR TITLE
Initial code commit

### PR DESCRIPTION
**Pull request by imjfckm**
Adding opencas-ansible stuff:
* example playbooks
* cas module
* cas action plugin
* tests

### CODE REVIEW

File: library/cas.py [hunk 1]
```diff
@@ -91,7 +91,7 @@
       cache_mode:
         description:
           - Caching mode for cache
-        choices: ['wt', 'wb', 'wa', 'pt']
+        choices: ['wt', 'wb', 'wa', 'pt', 'wi']
```
Comment (micrakow) [inline library/cas.py]: wo

Comment (imjfckm) [reply library/cas.py]: Done

File: library/opencas.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+    empty_config.write(cas_util.cas_config.default_location)
+
+    return True
+
+
+def stop(flush):
+    caches_list = cas_util.get_caches_list()
+
+    devices_count = len(caches_list)
+    if devices_count == 0:
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply library/opencas.py]: Yep, fixed

File: library/opencas.py [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+      cache_device:
+        description:
+          - Path to device to be used as a cache
+      id:
+        description:
+          - id of cache to be created
+      cache_mode:
+        description:
+          - Caching mode for cache
+        choices: ['wt', 'wb', 'wa', 'pt']
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply library/opencas.py]: Done

File: library/opencas.py [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply library/opencas.py]: Done

File: group_vars/opencas-nodes.yml.sample [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+# This is a sample configuration of Intel CAS deployment configuration
+
+# List of all cache devices configuration
+opencas_cache_devices:
+  - cache_device: /dev/nvme0n1       # path to device or partition
+    id: 1                            # id used to corelate cores with cache instances
+    cache_mode: wt                   # caching mode <wt, wb, wa, pt>
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply group_vars/opencas-nodes.yml.sample]: Done

File: group_vars/opencas-nodes.yml.sample [hunk 6]
```diff
@@ -0,0 +1,24 @@
+---
+# This is a sample configuration of Intel CAS deployment configuration
```
[the remark this answers is not in the document]
Comment (imjfckm) [reply group_vars/opencas-nodes.yml.sample]: Fixed

File: tests/test_ansible_module_01.py [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+
+
+@patch("opencas.get_cas_version")
+@patch("opencas.get_caches_list")
+@patch("cas.setup_module_object")
+def test_module_get_facts_installed(
+    mock_setup_module, mock_get_caches_list, mock_get_version
+):
+    mock_get_version.return_value = {
+        "Intel(R) CAS Kernel Module": "03.08.00.01131011",
```
Comment (micrakow) [inline tests/test_ansible_module_01.py]: Remove Intel

**Pull request merged**